src/utils.c

```diff
@@ -254,3 +257,95 @@ is_scanner_only_pref (const char *pref)
     return 1;
   return 0;
 }
+
+/** @brief Writes scripts stats into a file.
+ *
+ * @param buf String to write.
+ * @param scan_id Scan ID for the file name.
+ * @param mode 2 to create the file, 0 to append text to the file,
+ *             1 to finish the json list removing the trailing comma before
+ *             appending the last text in the buffer.
+ *
+ */
+void
+write_script_stats (const char *buf, const char *scan_id, int mode)
+{
+  char *p = NULL;
+  FILE *path = NULL;
+  if (!prefs_get ("report_scripts"))
+    return;
+  p =
+    g_strdup_printf ("%s/%s-stats.json", prefs_get ("report_scripts"), scan_id);
+
+  if (mode == 0)
+    path = fopen (p, "a");
+  else if (mode == 1)
+    path = fopen (p, "r+");
+  else if (mode == 2)
+    path = fopen (p, "w");
+
+  if (path == NULL)
```

Comment:
I think that hides possible usage issues.

I would double check the input mode before trying to open the given path:

```
  if (mode < 0 || mode > 2) {
    g_warning("invalid mode");
    return;
  }
  path = fopen (p, mode == 0 ? "a" : mode == 1 ? "r+" : "w");
  if (path == NULL) {
  ...
 }
```